lean4-infoview/src/infoview/info.tsx

```diff
@@ -123,8 +124,37 @@ export function InfoDisplay(props0: InfoDisplayProps) {
     const hasGoals = status !== 'error' && goals;
     const hasTermGoal = status !== 'error' && termGoal;
     const hasMessages = status !== 'error' && messages.length !== 0;
-    const filterClasses = 'link pointer mh2 dim codicon fr ' + (reverseOrder ? 'codicon-arrow-up' : 'codicon-arrow-down');
-    const sortButton = <a className={filterClasses} onClick={e => { setReverseOrder(!reverseOrder); }} title="reverse list"/>
+    const sortClasses = 'link pointer mh2 dim codicon fr ' + (goalFilters.reverse ? 'codicon-arrow-up' : 'codicon-arrow-down');
+    const sortButton = <a className={sortClasses} title="reverse list" onClick={e => {
+        setGoalFilters(s => {
+            return { reverse: !s.reverse, isType: s.isType, isInstance: s.isInstance }
+        } ); }
```

Comment:
```suggestion
        setGoalFilters(s => ({ ...s, reverse: !s.reverse })); }
```

Reply:
Thanks, I fixed all 3 places this was needed.